Problem1.java

```diff
@@ -1 +1,40 @@
+// A binary search based program 
+// to find the only missing number 
+// in a sorted array of distinct 
+// elements within limited range. 
+import java.io.*; 
+
+class Problem1 
+{ 
+static int search(int ar[], 
+				int size) 
+{ 
+	int a = 0, b = size - 1; 
+	int mid = 0; 
+	while ((b - a) > 1) 
+	{ 
+		mid = (a + b) / 2; 
```

Comment:
Instead of waiting for b-a to be =1, Can you thing of some early condition to terminate the loop?

Reply:
if(mid == a) break;

Reply:
This Solution wont work for the given test case.
Input : [1,3,4,5,6,7,8]